### PR TITLE
Exec create user

### PR DESCRIPTION
**Pull request by stevenewey**
The user parameter has been present but unimplemented in the code for a little while, I'm not sure if there's a good reason for that? Hopefully this is an good enough implementation of it.

### CODE REVIEW

File: tests/integration_test.py [hunk 1]
```diff
@@ -830,6 +830,23 @@ def runTest(self):
...
+        id = container['Id']
+        self.client.start(id)
+        self.tmp_containers.append(id)
+
+        res = self.client.exec_create(id, 'whoami', user='default')
+        self.assertIn('Id', res)
+
+        exec_log = self.client.exec_start(res)
+        expected = b'default' if six.PY3 else 'default\n'
+        self.assertEqual(exec_log, expected)
```
Comment (aanand) [inline tests/integration_test.py]: Could we also test that `whoami` returns `root` if `user` isn't passed in?

Comment (stevenewey) [inline tests/integration_test.py]: Added.

**Comment by aanand** [comment]
LGTM

**Comment by schu** [comment]
+1

File: docker/client.py [hunk 2]
```diff
@@ -521,19 +521,26 @@ def execute(self, container, cmd, detach=False, stdout=True, stderr=True,
...
             )
+        if user and utils.compare_version('1.19', self._version) < 0:
+            raise errors.InvalidVersion(
+                'User-specific exec is not supported in API < 1.19'
+            )
         if isinstance(cmd, six.string_types):
             cmd = shlex.split(str(cmd))
 
+        if user is None:
+            user = ''
```
Comment (shin-) [inline docker/client.py]: Can you explain the reasoning here? Seems unnecessary.

Comment (stevenewey) [inline docker/client.py]: You're right, it's not necessary, I'll remove it.

**Comment by shin-** [comment]
Awesome, thanks!

**Pull request merged**